src/csharp/Microsoft.Spark.E2ETest/UdfTests/BroadcastTests.cs

```diff
@@ -0,0 +1,107 @@
+        {
+            _spark = fixture.Spark;
+            var data = new List<string>(new string[] { "Alice", "Bob" });
+            _df = _spark.CreateDataFrame(data);
+        }
+
+        /// <summary>
+        /// Function tests Broadcast support by using a single broadcast variable in a UDF.
+        /// </summary>
+        [Fact]
+        public void TestSingleBroadcastWithoutEncryption()
+        {
+            var objectToBroadcast = new BroadcastTestClass(
+                1,
+                "testing first broadcast",
+                1.1,
+                true);
+
+            Broadcast bc = _spark.SparkContext.Broadcast(objectToBroadcast);
+
+            Func<Column, Column> testBroadcast = Udf<string, string>(
+                row => row + " is " + 
+                ((BroadcastTestClass)bc.Value()).stringValue + 
+                ", " + ((BroadcastTestClass)bc.Value()).intValue + 
+                ", " + ((BroadcastTestClass)bc.Value()).doubleValue + 
+                " => " + ((BroadcastTestClass)bc.Value()).boolValue);
+
+            string[] expected = new[] {
+                "Alice is testing first broadcast, 1, 1.1 => True",
+                "Bob is testing first broadcast, 1, 1.1 => True" };
+
+            DataFrame udfOutput = _df.Select(testBroadcast(_df["_1"]));
+            Row[] actualRows = udfOutput.Collect().ToArray();
+            string[] actual = actualRows.Select(s => s[0].ToString()).ToArray();
+            Assert.Equal(expected, actual);
+            bc.Unpersist();
+        }
+
+        /// <summary>
+        /// Function tests Broadcast support by using 2 broadcast variables in a UDF
+        /// </summary>
+        [Fact]
+        public void TestMultipleBroadcastWithoutEncryption()
+        {
+            var object1ToBroadcast = new BroadcastTestClass(
+                1,
+                "testing first broadcast",
+                1.1,
+                true);
+            var object2ToBroadcast = new BroadcastTestClass(
+                2,
+                "testing second broadcast",
+                2.2,
+                false);
+
+            Broadcast bc1 = _spark.SparkContext.Broadcast(object1ToBroadcast);
+            Broadcast bc2 = _spark.SparkContext.Broadcast(object2ToBroadcast);
+
+            Func<Column, Column> testBroadcast = Udf<string, string>(
+                row => row + " is " +
+                ((BroadcastTestClass)bc1.Value()).stringValue +
+                " and " + ((BroadcastTestClass)bc2.Value()).stringValue);
+
+            string[] expected = new[] {
+                "Alice is testing first broadcast and testing second broadcast",
+                "Bob is testing first broadcast and testing second broadcast" };
+
+            DataFrame udfOutput = _df.Select(testBroadcast(_df["_1"]));
+            Row[] actualRows = udfOutput.Collect().ToArray();
+            string[] actual = actualRows.Select(s => s[0].ToString()).ToArray();
+            Assert.Equal(expected, actual);
```

Comment:
Do we need to add `bc.Unpersist();` here as well?

Reply:
I can make a separate testcase for Unpersist and Destroy.